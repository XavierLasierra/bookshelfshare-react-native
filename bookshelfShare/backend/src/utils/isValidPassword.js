const bcrypt = require("bcrypt");

function isValidPassword(password) {
  return bcrypt.compareSync(password, this.password);
}

module.exports = { isValidPassword };
