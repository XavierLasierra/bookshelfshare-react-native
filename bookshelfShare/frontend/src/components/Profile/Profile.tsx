import React from 'react';
import {
  SafeAreaView, View, Image, Text, TouchableOpacity, ScrollView
} from 'react-native';
import { useSelector } from 'react-redux';
import Header from '../Header/Header';

import ProfileIcon from '../../assets/profileIcon.svg';
import BookStackIcon from '../../assets/bookStackIcon.svg';
import ShelfIcon from '../../assets/shelfIcon.svg';
import styles from './profile.styles';
import globalStyles from '../../styles/global.styles';

export default function Profile() {
  const { userData } = useSelector((store: any) => store.loggedUser);
  const shelves = useSelector((store: any) => store.shelves);

  function handleLogout() {

  }

  return (
    <SafeAreaView style={globalStyles.mainContainer}>
      <ScrollView>
        <Header Logo={ProfileIcon} />
        <View style={styles.profileContainer}>
          <View style={styles.topContainer}>
            <Image
              source={{ uri: userData.photo }}
              style={styles.userPhoto}
            />
            <View style={styles.userInformationContainer}>
              <Text style={styles.userUsername}>{userData.username}</Text>
              <Text style={styles.userEmail}>{userData.email}</Text>
            </View>

          </View>
          <View style={styles.actionsContainer}>
            <View style={styles.socialContainer}>
              <TouchableOpacity style={[styles.profileButton, styles.socialButton]}>
                <ProfileIcon width={25} height={25} />
                <Text style={styles.buttonText}>
                  {userData.followers.length}
                  {' '}
                  {userData.followers.length === 1 ? 'follower' : 'followers'}
                </Text>
              </TouchableOpacity>
              <TouchableOpacity style={[styles.profileButton, styles.socialButton]}>
                <ProfileIcon width={25} height={25} />
                <Text style={styles.buttonText}>
                  {userData.following.length}
                  {' '}
                  following
                </Text>
              </TouchableOpacity>
            </View>
            <TouchableOpacity style={[styles.profileButton, styles.booksButton]}>
              <BookStackIcon width={40} height={40} />
              <Text style={styles.buttonText}>
                {userData.books.current.length}
                {' '}
                {userData.books.current.length === 1 ? 'book' : 'books'}
                {' '}
                currently reading
              </Text>
            </TouchableOpacity>
            <TouchableOpacity style={[styles.profileButton, styles.booksButton]}>
              <BookStackIcon width={40} height={40} />
              <Text style={styles.buttonText}>
                {userData.books.read.length}
                {' '}
                {userData.books.read.length === 1 ? 'book' : 'books'}
                {' '}
                read
              </Text>
            </TouchableOpacity>
            <TouchableOpacity style={[styles.profileButton, styles.booksButton]}>
              <BookStackIcon width={40} height={40} />
              <Text style={styles.buttonText}>
                {userData.books.wishlist.length}
                {' '}
                {userData.books.wishlist.length === 1 ? 'book' : 'books'}
                {' '}
                on wishlist
              </Text>
            </TouchableOpacity>

            <TouchableOpacity style={[styles.profileButton, styles.booksButton]}>
              <ShelfIcon width={40} height={35} />
              <Text style={styles.buttonText}>
                {shelves.length}
                {' '}
                {shelves.length === 1 ? 'shelf' : 'shelves'}
              </Text>
            </TouchableOpacity>
          </View>
          <TouchableOpacity
            style={[globalStyles.button, styles.logoutButton]}
            onPress={handleLogout}
            testID="logoutButton"
          >
            <Text style={globalStyles.buttonText}>Log out</Text>
          </TouchableOpacity>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
}
