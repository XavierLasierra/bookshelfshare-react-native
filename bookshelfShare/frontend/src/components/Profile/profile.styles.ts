import { StyleSheet } from 'react-native';
import stylesConstants from '../../styles/styles.constants';

export default StyleSheet.create({
  actionsContainer: {
    marginTop: 20
  },
  socialContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between'
  },
  socialButton: {
    flexDirection: 'row',
    alignItems: 'center',
    borderTopWidth: 2,
    borderBottomWidth: 2,
    borderColor: stylesConstants.colors.secondary,
    marginBottom: 15,
    height: 50,
    width: 160,
    paddingHorizontal: 10
  },
  buttonText: {
    fontSize: stylesConstants.fontSize.mid,
    marginLeft: 15
  },
  logoutButton: {
    marginTop: 50
  },
  marginBottom: {
    width: '100%',
    height: 130
  }
});
