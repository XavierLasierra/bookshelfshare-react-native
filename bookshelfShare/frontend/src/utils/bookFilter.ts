export function bookFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfListFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.bookData.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfLocationFilter(filterLocation: number[], books: any) {
  return books.filter(({ customInformation: { location } }: any) => JSON.stringify(location)
  === JSON.stringify(filterLocation));
}
