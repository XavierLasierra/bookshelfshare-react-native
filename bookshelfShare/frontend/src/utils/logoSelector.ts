import BookIcon from '../assets/bookIcon.svg';
import SearchIcon from '../assets/searchIcon.svg';
import ProfileIcon from '../assets/profileIcon.svg';

export default function logoSelector(logoName: string) {
  switch (logoName) {
    case 'SearchIcon':
      return SearchIcon;
    case 'BookIcon':
      return BookIcon;
    case 'ProfileIcon':
      return ProfileIcon;
    default:
      return BookIcon;
  }
}
