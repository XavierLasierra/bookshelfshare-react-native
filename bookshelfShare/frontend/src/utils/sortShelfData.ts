export default function sortShelfData(shelf: any) {
  return shelf?.books.reduce((acc: any, book: any) => {
    const foundShelfBox = acc.find(({ location }: any) => JSON.stringify(location)
         === JSON.stringify(book?.customInformation?.location));
    if (foundShelfBox) {
      foundShelfBox.books.push(book.bookData);
    } else {
      acc.push({
        location: book?.customInformation?.location,
        books: [book.bookData]
      });
    }
    return acc;
  }, []);
}
