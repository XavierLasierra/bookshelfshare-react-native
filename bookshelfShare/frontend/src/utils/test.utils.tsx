import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import configureStore from '../redux/store';

function render(
  component: any,
  initialState: any
) {
  const store = configureStore(initialState);

  function Wrapper({ children }: any) {
    return (
      <Provider store={store}>
        {children}
      </Provider>
    );
  }
  return rtlRender(component, { wrapper: Wrapper });
}

// re-export everything
export * from '@testing-library/react-native';
// override render method
export { render };
