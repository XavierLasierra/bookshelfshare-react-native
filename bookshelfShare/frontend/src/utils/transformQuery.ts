interface Query {
    isbn?: string,
    inauthor?: string,
    intitle?: string,
    inpublisher?: string
}

export default function transformQuery(query: Query):string {
  const queryEntries = Object.entries(query);
  const url = queryEntries
    .reduce((acc, queryElement, index) => `${acc}${queryElement[0]}=${queryElement[1]}${index < queryEntries.length - 1 ? '+' : ''}`, '');
  return `${url}&key=${process.env.GOOGLE_API_KEY}`;
}
