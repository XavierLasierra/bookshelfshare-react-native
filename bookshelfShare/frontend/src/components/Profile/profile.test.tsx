import React from 'react';
import Profile from './Profile';
import {render, fireEvent} from '../../utils/test.utils';
import loggedUserMock from '../../mocks/loggedUser.mock';
import userBooksMock from '../../mocks/userBooks.mock';
import userSocialsMock from '../../mocks/userSocials.mock';
import {logoutUser} from '../../redux/actions/loggedUser.creator';
import {getBooksData} from '../../redux/actions/books.creator';
import navigationMock from '../../mocks/navigation.mock';

jest.mock('../../redux/actions/loggedUser.creator');
jest.mock('../../redux/actions/books.creator');

describe('Given a Profile component', () => {
  describe('When it is rendered', () => {
    describe('And the user has no followers', () => {
      test('Then should match the snapshot', () => {
        const initialState = {
          loggedUser: {userData: loggedUserMock.user},
          userBooks: userBooksMock,
        };
        const screen = render(
          <Profile navigation={navigationMock} />,
          initialState,
        );
        expect(screen).toMatchSnapshot();
      });
    });

    describe('And the user has exactly 1 follower', () => {
      let screen: any;
      let navigation: any;
      beforeEach(() => {
        (logoutUser as jest.Mock).mockReturnValue({type: ''});
        (getBooksData as jest.Mock).mockReturnValue({type: ''});

        const initialState = {
          loggedUser: {userData: loggedUserMock.user},
          userBooks: userBooksMock,
          userSocials: userSocialsMock,
        };
        navigation = {
          ...navigationMock,
          push: jest.fn(),
        };
        screen = render(<Profile navigation={navigation} />, initialState);
      });
      test('Then should match the snapshot', () => {
        expect(screen).toMatchSnapshot();
      });

      describe('And you press followersButton', () => {
        test('Then navigation.push should have been called', () => {
          const followersButton = screen.getByTestId('followersButton');
          fireEvent.press(followersButton);

          expect(navigation.push).toHaveBeenCalled();
        });
      });

      describe('And you press followingButton', () => {
        test('Then navigation.push should have been called', () => {
          const followingButton = screen.getByTestId('followingButton');
          fireEvent.press(followingButton);

          expect(navigation.push).toHaveBeenCalled();
        });
      });

      describe('And you press logoutButton', () => {
        test('Then logoutUser function should have been called', () => {
          const logoutButton = screen.getByTestId('logoutButton');
          fireEvent.press(logoutButton);

          expect(logoutUser).toHaveBeenCalled();
        });
      });

      describe('And you click in one of the books lists', () => {
        beforeEach(() => {
          const bookListButton = screen.getByTestId('readButton');
          fireEvent.press(bookListButton);
        });
        test('Then getBooksData should have been called', () => {
          expect(getBooksData).toHaveBeenCalled();
        });
        test('Then navigation.push should have been called', () => {
          expect(navigation.push).toHaveBeenCalled();
        });
      });
    });
  });
});
