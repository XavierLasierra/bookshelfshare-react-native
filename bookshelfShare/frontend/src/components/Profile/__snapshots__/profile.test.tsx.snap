// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Profile component When it is rendered And the user has exactly 1 follower Then should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#F5F5F5",
      "minHeight": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#F5F5F5",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D9D8C4",
          "borderBottomLeftRadius": 25,
          "justifyContent": "center",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <IconMock
        height={45}
        width={45}
      />
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#D9D8C4",
            "height": 20,
            "width": 20,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopRightRadius": 25,
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "paddingHorizontal": 30,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "photo",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 125,
                "marginRight": 30,
                "width": 125,
              }
            }
          />
          <View
            style={
              Object {
                "maxWidth": 200,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "marginBottom": 15,
                }
              }
            >
              Xavier Lasierra
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 16,
                }
              }
            >
              xlp1998@gmail.com
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "marginTop": 20,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "borderBottomWidth": 2,
                  "borderColor": "#ECECEC",
                  "borderTopWidth": 2,
                  "flexDirection": "row",
                  "height": 50,
                  "marginBottom": 15,
                  "opacity": 1,
                  "paddingHorizontal": 10,
                  "width": 160,
                }
              }
              testID="followersButton"
            >
              <IconMock
                height={25}
                width={25}
              />
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "marginLeft": 15,
                  }
                }
              >
                1
                 
                follower
              </Text>
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "borderBottomWidth": 2,
                  "borderColor": "#ECECEC",
                  "borderTopWidth": 2,
                  "flexDirection": "row",
                  "height": 50,
                  "marginBottom": 15,
                  "opacity": 1,
                  "paddingHorizontal": 10,
                  "width": 160,
                }
              }
              testID="followingButton"
            >
              <IconMock
                height={25}
                width={25}
              />
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "marginLeft": 15,
                  }
                }
              >
                1
                 
                following
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 2,
                "borderColor": "#ECECEC",
                "borderTopWidth": 2,
                "flexDirection": "row",
                "height": 50,
                "marginBottom": 15,
                "opacity": 1,
              }
            }
            testID="readingButton"
          >
            <IconMock
              height={40}
              width={40}
            />
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "marginLeft": 15,
                }
              }
            >
              1
               
              book
               
              currently reading
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 2,
                "borderColor": "#ECECEC",
                "borderTopWidth": 2,
                "flexDirection": "row",
                "height": 50,
                "marginBottom": 15,
                "opacity": 1,
              }
            }
            testID="readButton"
          >
            <IconMock
              height={40}
              width={40}
            />
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "marginLeft": 15,
                }
              }
            >
              1
               
              book
               
              read
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 2,
                "borderColor": "#ECECEC",
                "borderTopWidth": 2,
                "flexDirection": "row",
                "height": 50,
                "marginBottom": 15,
                "opacity": 1,
              }
            }
            testID="toReadButton"
          >
            <IconMock
              height={40}
              width={40}
            />
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "marginLeft": 15,
                }
              }
            >
              1
               
              book
               
              to read
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 2,
                "borderColor": "#ECECEC",
                "borderTopWidth": 2,
                "flexDirection": "row",
                "height": 50,
                "marginBottom": 15,
                "opacity": 1,
              }
            }
            testID="wishlistButton"
          >
            <IconMock
              height={40}
              width={40}
            />
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "marginLeft": 15,
                }
              }
            >
              1
               
              book
               
              on wishlist
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={false}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 2,
                "borderColor": "#ECECEC",
                "borderTopWidth": 2,
                "flexDirection": "row",
                "height": 50,
                "marginBottom": 15,
                "opacity": 1,
              }
            }
          >
            <IconMock
              height={35}
              width={40}
            />
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "marginLeft": 15,
                }
              }
            >
              0
               
              shelves
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#000000",
              "borderRadius": 10,
              "display": "flex",
              "height": 70,
              "justifyContent": "center",
              "marginTop": 50,
              "opacity": 1,
              "width": "100%",
            }
          }
          testID="logoutButton"
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 20,
                "fontWeight": "500",
              }
            }
          >
            Log out
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 130,
            "width": "100%",
          }
        }
      />
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`Given a Profile component When it is rendered And the user has no followers Then should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#F5F5F5",
      "minHeight": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#F5F5F5",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D9D8C4",
          "borderBottomLeftRadius": 25,
          "justifyContent": "center",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <IconMock
        height={45}
        width={45}
      />
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#D9D8C4",
            "height": 20,
            "width": 20,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopRightRadius": 25,
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "paddingHorizontal": 30,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "photo",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 125,
                "marginRight": 30,
                "width": 125,
              }
            }
          />
          <View
            style={
              Object {
                "maxWidth": 200,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "marginBottom": 15,
                }
              }
            >
              Xavier Lasierra
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 16,
                }
              }
            >
              xlp1998@gmail.com
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "marginTop": 20,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "borderBottomWidth": 2,
                  "borderColor": "#ECECEC",
                  "borderTopWidth": 2,
                  "flexDirection": "row",
                  "height": 50,
                  "marginBottom": 15,
                  "opacity": 1,
                  "paddingHorizontal": 10,
                  "width": 160,
                }
              }
              testID="followersButton"
            >
              <IconMock
                height={25}
                width={25}
              />
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "marginLeft": 15,
                  }
                }
              >
                0
                 
                followers
              </Text>
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "borderBottomWidth": 2,
                  "borderColor": "#ECECEC",
                  "borderTopWidth": 2,
                  "flexDirection": "row",
                  "height": 50,
                  "marginBottom": 15,
                  "opacity": 1,
                  "paddingHorizontal": 10,
                  "width": 160,
                }
              }
              testID="followingButton"
            >
              <IconMock
                height={25}
                width={25}
              />
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "marginLeft": 15,
                  }
                }
              >
                0
                 
                following
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 2,
                "borderColor": "#ECECEC",
                "borderTopWidth": 2,
                "flexDirection": "row",
                "height": 50,
                "marginBottom": 15,
                "opacity": 1,
              }
            }
            testID="readingButton"
          >
            <IconMock
              height={40}
              width={40}
            />
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "marginLeft": 15,
                }
              }
            >
              1
               
              book
               
              currently reading
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 2,
                "borderColor": "#ECECEC",
                "borderTopWidth": 2,
                "flexDirection": "row",
                "height": 50,
                "marginBottom": 15,
                "opacity": 1,
              }
            }
            testID="readButton"
          >
            <IconMock
              height={40}
              width={40}
            />
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "marginLeft": 15,
                }
              }
            >
              1
               
              book
               
              read
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 2,
                "borderColor": "#ECECEC",
                "borderTopWidth": 2,
                "flexDirection": "row",
                "height": 50,
                "marginBottom": 15,
                "opacity": 1,
              }
            }
            testID="toReadButton"
          >
            <IconMock
              height={40}
              width={40}
            />
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "marginLeft": 15,
                }
              }
            >
              1
               
              book
               
              to read
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 2,
                "borderColor": "#ECECEC",
                "borderTopWidth": 2,
                "flexDirection": "row",
                "height": 50,
                "marginBottom": 15,
                "opacity": 1,
              }
            }
            testID="wishlistButton"
          >
            <IconMock
              height={40}
              width={40}
            />
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "marginLeft": 15,
                }
              }
            >
              1
               
              book
               
              on wishlist
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={false}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 2,
                "borderColor": "#ECECEC",
                "borderTopWidth": 2,
                "flexDirection": "row",
                "height": 50,
                "marginBottom": 15,
                "opacity": 1,
              }
            }
          >
            <IconMock
              height={35}
              width={40}
            />
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "marginLeft": 15,
                }
              }
            >
              0
               
              shelves
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#000000",
              "borderRadius": 10,
              "display": "flex",
              "height": 70,
              "justifyContent": "center",
              "marginTop": 50,
              "opacity": 1,
              "width": "100%",
            }
          }
          testID="logoutButton"
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 20,
                "fontWeight": "500",
              }
            }
          >
            Log out
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 130,
            "width": "100%",
          }
        }
      />
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
